src/ansys/aedt/core/modeler/circuits/primitives_twin_builder.py

```diff
@@ -475,3 +475,111 @@ def create_periodic_waveform_source(
             id.set_property("PERIO", 1)
 
         return id
+
+    @pyaedt_function_handler()
+    def create_component_from_sml(
+        self,
+        input_file,
+        model,
+        pins_names,
+    ):
+        """Create and place a new component based on a .sml file.
+
+        Parameters
+        ----------
+        input_file : str
+            Path to .sml file.
+        model : str
+            Model name to import.
+        pins_names : str
```

Comment:
Should the type be a list?

Reply:
There is a call to `pins_names.lower()` so I assume that's not the case :)

Reply:
Both of you are right, I like more a list. I will change it, because it is more pythonic.

I have implemented it as a string because it is what AEDT API expects, but we can do that conversion inside

Reply:
Sounds good to me.